bokehjs/src/coffee/core/has_props.coffee

```diff
@@ -368,22 +368,24 @@ class HasProps extends Backbone.Model
     _.values(result)
 
   attach_document: (doc) ->
-    if @document != null and @document != doc
-      throw new Error("Models must be owned by only a single document")
-    first_attach = @document == null
-    @document = doc
-    if doc != null
-      doc._notify_attach(@)
-      if first_attach
-        for c in @_immediate_references()
-          c.attach_document(doc)
+    if @document != null
+      if @document != doc
+        throw new Error("models must be owned by only a single document")
+      else
+        @document._notify_attach(@)
+    else
+      @document = doc
+      @document._notify_attach(@)
+
+      for ref in @_immediate_references()
+        ref.attach_document(@document)
 
-    # TODO (bev) is there are way to get rid of this?
-    for name, prop of @properties
-      prop.update()
+      # TODO (bev) is there are way to get rid of this?
+      for name, prop of @properties
+        prop.update()
```

Comment:
@bryevdv, I see. This explanation raises my attention, so I will look into this.